File: FileReadWrite/app/src/main/java/com/example/filereadwrite/MainActivity.java
package com.example.filereadwrite;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.InputStreamReader;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        EditText editText= findViewById(R.id.editText);
        Button btnWrite= findViewById(R.id.btnWrite);
        Button btnRead= findViewById(R.id.btnRead);
        TextView textView=findViewById(R.id.textView);
        btnWrite.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String text = editText.getText().toString();

                File dataDir = ContextCompat.getDataDir(MainActivity.this);
                File myFile = new File(dataDir,"file.txt");

                try {
                    FileOutputStream fos = new FileOutputStream(myFile);
                    fos.write(text.getBytes());
                } catch (FileNotFoundException e) {
                    Toast.makeText(MainActivity.this, "file not found", Toast.LENGTH_SHORT).show();
                } catch (IOException e) {
                    Toast.makeText(MainActivity.this, "Error while writing file", Toast.LENGTH_SHORT).show();
                }

            }
        });
        /* In Kotlin
        btnWrite.setOnClickListener{
        val dataDir = ContextCompat.getDataDir(this)
        val myFile = File(dataDir, "file.txt")
        myFile.writeText(editText.text.toString())
        }

        btnRead.setOnClickListener{
        val dataDir = ContextCompat.getDataDir(this)
        val myFile = File(dataDir, "file.txt")
        textView.text= myFile.readText()
        }
         */

        btnRead.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                File dataDir = ContextCompat.getDataDir(MainActivity.this);
                File myFile = new File(dataDir,"file.txt");

                try {
                    FileInputStream fis = new FileInputStream(myFile);
                    InputStreamReader inputStreamReader=new InputStreamReader(fis);
                    BufferedReader bufferedReader= new BufferedReader(inputStreamReader);

                    StringBuilder stringBuilder= new StringBuilder();
                    String buffer= bufferedReader.readLine();
                    while(buffer != null)
                    {
                        stringBuilder.append(buffer);
                        buffer= bufferedReader.readLine();
                    }

                    String text = stringBuilder.toString();
                    textView.setText(text);

                } catch (FileNotFoundException e) {
                    Toast.makeText(MainActivity.this, "File not found", Toast.LENGTH_SHORT).show();
                } catch (IOException e) {
                    Toast.makeText(MainActivity.this, "Error while reading", Toast.LENGTH_SHORT).show();
                }
            }
        });
    }
}

File: permission/app/src/main/java/com/example/permission/MainActivity.kt
package com.example.permission

import android.content.Context
import android.content.Intent
import android.net.ConnectivityManager
import android.net.NetworkInfo
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.core.app.ActivityCompat
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        button.setOnClickListener {
             textView.text= if (isNetworkAvailable()) "connected" else "notconnected"
        }
        button2.setOnClickListener {
            val i=Intent(this,MainActivity2::class.java)
            startActivity(i)
        }

    }
    private fun isNetworkAvailable(): Boolean {
        val cm=getSystemService(Context.CONNECTIVITY_SERVICE)
        return if (cm is ConnectivityManager){
            val networkInfo:NetworkInfo?=cm.activeNetworkInfo
            networkInfo?.isConnected?: false
        }
        else false
    }
}
/*  button.setOnClickListener(new View.OnClickListener(){
@Override
public void onClick(View v){
ConnectivityManager cm=(ConnectivityManager)getSystemService(CONNECTIVITY_SERVICE);
NetworkInfo networkInfo=cm.getActiveNetworkInfo();
boolean isConnected=networkInfo=!null && networkInfo.isConnected();
textView.setText(isConnected ? "Connected" : "Doisconnected");
}
});
IN JAVA*/
